src/main/java/com/rog/EShop/services/UserService.java

```diff
@@ -38,35 +42,91 @@ public UserDto findById(Long id) {
         return userMapper.toDTO(user);
     }
 
+
     public UserDto save(UserRegisterDto userRegisterDto) {
 
         User user = userMapper.toEntity(userRegisterDto);
         if (userRepository.existsUserByUsername(user.getUsername())) {
             throw new ConflictException("This username already exists!");
         }
-
         if (!Objects.equals(userRegisterDto.getPassword(), userRegisterDto.getConfirmPassword())) {
             throw new BadRequestException("Password does not match!");
         }
-        String encode = bCryptPasswordEncoder.encode(userRegisterDto.getPassword());
-        user.setPassword(encode);
-        user.setRegisterDate(LocalDateTime.now());
-        user.setAccountNonExpired(true);
-        user.setAccountNonLocked(true);
-        user.setCredentialsNonExpired(true);
-        user.setEnabled(true);
-        user.setAuthorities(List.of(Role.ROLE_USER));
-        User userSaved = userRepository.save(user);
-        return userMapper.toDTO(userSaved);
 
+        ResponseTokenDto responseTokenDto = getResponseTokenDto();
+        ResponseEntity<String> response = getStringResponseEntity(userRegisterDto, responseTokenDto);
+        if (response.getStatusCode().isError()) {
+            throw new RuntimeException("Error " + response.getStatusCode());
+        }
 
+        ResponseEntity<List<ResponseKeycloakDto>> responseKeycloakDto = getListResponseEntity(userRegisterDto, responseTokenDto);
+
+        List<ResponseKeycloakDto> responseKeycloakDto1 = responseKeycloakDto.getBody();
+        user.setKeycloakId(responseKeycloakDto1.get(0).getId());
+        Long timeStamp = responseKeycloakDto1.get(0).getCreatedTimestamp();
+        Instant instant = Instant.ofEpochMilli(timeStamp);
+        LocalDateTime localDate = LocalDateTime.ofInstant(instant, ZoneOffset.UTC);
+        user.setRegisterDate(localDate);
+        User user2 = userRepository.save(user);
+        return userMapper.toDTO(user2);
+     
     }
 
-    @Override
-    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
-        return userRepository.findByUsername(username)
-                .orElseThrow(() -> new UsernameNotFoundException("The user " + username + " does not exist"));
+    private static ResponseEntity<List<ResponseKeycloakDto>> getListResponseEntity(UserRegisterDto userRegisterDto, ResponseTokenDto responseTokenDto) {
+        RestTemplate restTemplate2 = new RestTemplate();
+        String url = "http://localhost:8080/admin/realms/ESTORE/users?username={username}&exact=true";
```

Comment:
hardcoded host `http://localhost:8080` is bad. You need to use Configuration Properties. 
Create separate class `ApplicationProperties.java` and set all your custom properties in `application.yaml` file.
Read more: https://www.baeldung.com/configuration-properties-in-spring-boot
replace `http://localhost:8080` in whole project using your custom properties class